packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -2171,6 +2172,11 @@ function commitMutationEffectsOnFiber(
           }
         }
       }
+      if (enableEagerAlternateStateNodeCleanup) {
```

Comment:
This should be in the else branch of `if (supportsMutation)` and it should check `if (supportsPersistence)` since this should only apply to the persistent mode which is the only mode that clones the state nodes.

The flag is only on in RN but it's still on for Paper unnecessarily. You're better off turning the flag on everywhere else because that flushes out any issues where it would have issues if it turned on in the other renders. Like not gating it properly on `supportsPersistence`.

Reply:
let me do that. My thinking was that it is a no-op in `supportsMutation` mode as stateNode will be the same between finishedWork and alternate. 

thanks for pointing out the `supportsPersistence`. I will also add a comment explaining why this is needed.